addons/website/static/src/snippets/s_alert/options.js

```diff
@@ -0,0 +1,30 @@
+/** @odoo-module **/
+
+import options from "@web_editor/js/editor/snippets.options";
+
+options.registry.Alert = options.Class.extend({
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Changes alert's icon pictogram.
+     *
+     * @see this.selectClass for parameters
+     */
+    selectAlertIcon(previewMode, widgetValue, params) {
+        const alertIconEl = this.$target[0].querySelector('.s_alert_icon');
+        if (!alertIconEl) {
+            return;
+        }
+
+        // Note: this function is basically a "selectClass" combined with an
+        // "applyTo" but each option already comes with a "selectClass"
+        // targeting the main container. This is also why this does not need
+        // a _computeWidgetState, relying on the "selectClass" one which comes
+        // afterwards alphabetically (compared to "selectAlertIcon").
+        alertIconEl.classList.remove(...params.possibleValues.filter(v => !!v));
```

Comment:
@bvr-odoo It seems that this class removal doesn't handle all possible scenarios... If we remove only the `possibleValues` and we set a different icon from the media dialog, we will end up with an element holding more than one `fa-` class, which leads to inconsistencies :grimacing:

https://github.com/odoo/odoo/assets/25032876/750baed1-b7a6-4f53-8a60-185e6acea6a4

We simply need to remove all icons classes before setting the right one :+1: